torch/fx/_symbolic_trace.py

```diff
@@ -853,13 +853,15 @@ def wrapped(*args, **kwargs):
         # import here to avoid circular imports
         from .experimental.proxy_tensor import get_innermost_proxy_mode, proxy_call, disable_proxy_modes_tracing
 
-        # If there is no input with proxy, see if we are tracing with proxy tensors
-        proxy_mode = get_innermost_proxy_mode()
-        if proxy_mode is not None:
-            # Disable tracing of the interior of the wrapped fn while evaluating
-            with disable_proxy_modes_tracing():
-                out = proxy_call(proxy_mode, orig_fn, args, kwargs)
-            return out
+        # Check if we want to trace proxy tensors created via `make_fx`,
+        if visible_to_make_fx:
```

Comment:
I still feel strange regarding #94461 (and the fix here). If `make_fx` requires some special handling during `torch.fx.symbolic_trace`, it can implement its own tracer. I am not sure if patching the base mechanism is a sustainable idea.

Reply:
Well, I am in partial agreement. 

One option is for `wrap(visible_to_make_fx=True)` to fill up its own table and perform its own wrapping. 

However, I feel that it is overly cumbersome to do so, and will result in a lot of duplication and complexity introduced back into `make_fx`. I think that since `make_fx` is quite canonical, supporting just this case is fine.

Reply:
Code for `proxy_tensor.py` is in fact not bad. It helps a lot when I need to make proxy_tensor specific hacks. Tangling up code for fx.symbolic_trace and proxy_tensor could make my life way harder. Any, just log my concern here --- I'd avoid any confusion code if possible or write "enough" document. Not want block PR.